src/nix/utils.sh

```diff
@@ -282,19 +282,26 @@ find_prev_version_from_git_tags() {
     major="$(echo "${current_version}" | grep -oE '^[0-9]+' || echo '')"
     minor="$(echo "${current_version}" | grep -oP '^[0-9]+\.\K[0-9]+' || echo '')"
     breakfix="$(echo "${current_version}" | grep -oP '^[0-9]+\.[0-9]+\.\K[0-9]+' 2>/dev/null || echo '')"
-    set -e
+
+    if [ "${minor}" = "0" ] && [ "${breakfix}" = "0" ]; then
+        ((major=major-1))
+        declare -g ${variable_name}="${major}"
+        # Look for latest version from previous major release
+        find_version_from_git_tags "${variable_name}" "${repository}" "${prefix}" "${separator}" "${last_part_optional}"
     # Handle situations like Go's odd version pattern where "0" releases omit the last part
-    if [ "${breakfix}" = "" ] || [ "${breakfix}" = "0" ]; then
```

Comment:
Was the change in the `nix` Feature intentional?  Curious if @Chuxel intended for this condition to be separate?

Reply:
Go omits the last digit for major versions.  It's 1.19 not 1.19.0, so yes, that was intentional.

Reply:
> Was the change in the nix Feature intentional? 

Yep, it was made to handle major version changes (eg. previous version from `2.0.0` for git is `1.14.1`)